lib/resty/memcached.lua

```diff
@@ -193,26 +196,48 @@ function _M.get(self, key)
     end
 
     -- print("len: ", len, ", flags: ", flags)
-
-    local data, err = sock:receive(len)
-    if not data then
-        if err == "timeout" then
-            sock:close()
-        end
+    local data, err = _read_reply(sock, len)
+    if err then
         return nil, nil, err
     end
 
-    line, err = sock:receive(7) -- discard the trailing "\r\nEND\r\n"
-    if not line then
-        if err == "timeout" then
-            sock:close()
-        end
+    local _, err = _read_reply(sock, 7) -- discard the trailing "\r\nEND\r\n"
+    if err then
         return nil, nil, err
     end
 
     return data, flags
 end
 
+function _M.get(self, key)
+    if type(key) == "table" then
+        return _multi_get(self, key)
+    end
+
+    local sock = self.sock
+    if not sock then
+        return nil, nil, "not initialized"
+    end
+
+    local reqs = rawget(self, "_reqs")
+    if reqs then
+        local readers = rawget(self, "_readers")
+        tab_insert(reqs, "get ")
+        tab_insert(reqs, self.escape_key(key))
+        tab_insert(reqs, "\r\n")
+        tab_insert(readers, _get_reply)
+        return 1
+    end
+
+    local req = {"get ", self.escape_key(key), "\r\n"}
```

Comment:
I'm not sure it is worth creating a new table to avoid concating a new string in Lua Land.
The table can be reused and we'd better do some benchmark tests when doing such changes.

Reply:
The performance difference between the two implementations is not significant, but the new scheme should use less memory

Reply:
Maybe we should reuse the table?